lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -174,6 +174,11 @@ def add_resource_relationships(attrs, serializer, options = {})
               end
             end
 
+            if association.respond_to?(:json_api_links) &&
+               @options[:include].include?(name.to_s)
+              add_links(attrs[:relationships][name], association)
```

Comment:
I don't think this is correct. Links on a relationships might be different than links on a resource. For example, on a relationship we might have `"self": "/articles/1/relationships/tags"` and on the resource we might have `"self": "/tags/1"`.